### PR TITLE
Add all ecr_repository args from AWS provider 2.x

### PR DESCRIPTION
**Pull request by bacek**

### CODE REVIEW

File: tools/gen-providers.ts [hunk 1]
```diff
@@ -610,9 +611,33 @@ const iam_group_policy: RecordDecl = {
...
+    requiredField('scan_on_push', BOOLEAN),
+  ],
+};
+
 const ecr_repository: RecordDecl = {
   name: 'ecr_repository',
-  fields: [requiredField('name', STRING)],
+  fields: [
+    requiredField('name', STRING),
+    // AWS 3.x encryption_configuration options
```
Comment (paul-thompson-helix) [inline tools/gen-providers.ts:635]:
```suggestion
    // AWS 2.x encryption_configuration options
```

**Comment by paul-thompson-helix** [comment]
LGTM

image_tag_mutability I've used - handy for enforcing image tags to be immutable (but does prevent re-use of the 'latest' tag however)...

Are you setting up image encryption and/or image scanning?

**Review by paul-thompson-helix — APPROVED** [review]

**Comment by bacek** [comment]
> LGTM
> 
> image_tag_mutability I've used - handy for enforcing image tags to be immutable (but does prevent re-use of the 'latest' tag however)...

tag immutability is the main reason for this PR.

> Are you setting up image encryption and/or image scanning?

Nope. And encryption is AWS provider 3.x. So until we update tf to less ancient version it's impossible to use. I just added it for future proofing.

File: tools/gen-providers.ts (hunk 1 above)
Comment (bacek) [reply tools/gen-providers.ts:635]: It is 3.x only option.

**Pull request merged**